### DISCUSSION TITLE
Website (Caddy) & Pi Monitoring Options?

### DISCUSSION THREAD

**Discussion by domenicdistefano** [mcguirepr89/BirdNET-Pi#993] [General]
Hi!

First let me say wow, By accident I found this project and couldn't believe how easy it was to get everything working. The single line of code to do the installation was just stupid easy. In just 24 hours I have identified 17 species and recorded 363 songs.

Things worked out so well I "donated" the installation / site to my local neighborhood. I am running the Pi in my attic with the microphone through a vent. I registered a domain name and am proxying it through my Synology NAS (NGINX proxy) with a Lets Encrypt cert. Everything is working perfectly. Once I announce this to my community I am sure I will have plenty of neighbors enjoying  it.

So now I want to do some basic hardware monitoring (attic gets hot) + performance trending and basic website statistics such as users per day, week, month.

Any suggestions? I haven't done much  research yet but I suspect there are plenty of hardware performance monitoring apps floating around for the Pi itself. I did however look for something that will provide analytics for Caddy and came up with very little. I did find [this](https://github.com/ldobbelsteen/caddy-analytics) but I am not sure how or if it would work. 

Any ideas?

Thanks

**Comment by jungle-boogie** [comment]
> So now I want to do some basic hardware monitoring (attic gets hot) + performance trending and basic website statistics such as users per day, week, month.

So you want to know how busy the system is over a given week? Honestly, unless you're having the pi do something else, I doubt you would see much of a change from week-to-week.

You could periodically login to the pi and run `top` or `htop` to take a peak at its activity.

Now if you want to know if the pi goes offline, you could setup a cronjob and be alerted if it stops responding.

Check out https://healthchecks.io/

**Comment by domenicdistefano** [comment]
I use [Uptime Kuma](https://uptime.kuma.pet/) to do basic ping uptime monitoring and alerting for devices around the house already. Since my Birdnet Pi is in the attic (gets HOT in the summer) it would be nice to see the trending of the hardware temps plus long term CPU and SSD / storage stats just for reference. 

As far as the web stats go I am not worried about load but just curious about how many of my neighbors will be using it on a regular basis.

**Comment by jungle-boogie** [comment]
I guess i just don't see a benefit in knowing the raspberry pi temperature, unless I was to act on it by turning it off.
You already know it's hot, so maybe you should consider ways to keep it cooler.

**Comment by domenicdistefano** [comment]
I solved the hardware monitoring! 

https://github.com/oijkn/Docker-Raspberry-PI-Monitoring

Took about 10 commands and had all of this running on the same Pi as Birdnet in about five minutes. Also note I am booting off of an SSD instead of an SD card. Simple / trivial thing to do and way more reliable. 

Blow is a screenshot of the monitoring dashboard


![image](https://github.com/mcguirepr89/BirdNET-Pi/assets/7042586/4439a8f0-adae-4965-8c43-ea510900a1a0)
